### ISSUE TITLE
Question on Pre-Training Output and Loss Function in Patch-Based Prediction

### ISSUE THREAD

**Issue by aohu1105** [yxbian23/aLLM4TS#2]
I have a question regarding the pre-training process. If the input consists of the 1st to 20th patches, should the output of pre-training be the 2nd to 21st patches? If this understanding is correct, I was wondering why the loss function uses **_batch_y_**, which seems to be the patches within the prediction window.

To provide more context: if the input window is 128 and patch_len = 16, without considering stride (simplifying stride to 16), the input would be 8 patches. After pre-training, the predicted output should be the last seven patches of the input window, plus the first patch of the prediction window. However, from your code, it seems that batch_y is divided into a fixed number of patches and used as the ground truth, which feels incorrect to me.

**Comment by yxbian23** [comment]
> I have a question regarding the pre-training process. If the input consists of the 1st to 20th patches, should the output of pre-training be the 2nd to 21st patches? If this understanding is correct, I was wondering why the loss function uses **_batch_y_**, which seems to be the patches within the prediction window.
> 
> To provide more context: if the input window is 128 and patch_len = 16, without considering stride (simplifying stride to 16), the input would be 8 patches. After pre-training, the predicted output should be the last seven patches of the input window, plus the first patch of the prediction window. However, from your code, it seems that batch_y is divided into a fixed number of patches and used as the ground truth, which feels incorrect to me.

1. Your understanding is correct. If the input consists of the **1st to 20th** patches, the output of pre-training should be the **2nd to 21st** patches.
2. However, you may have overlooked [the details of the construction of batch_y](https://github.com/yxbian23/aLLM4TS/blob/10b05be3be43ead40aec027b9c2e613153367aed/data_provider/data_loader.py#L121) before it is used as a label. **Batch_y is ahead of batch_x by a stride**, so we can input 1st to n-th patches and predict 2 to (n+1)-th patches.

**Comment by aohu1105** [comment]
Thank you very much for answering my question
